addons/website/static/src/js/content/inject_dom.js

```diff
@@ -15,6 +15,25 @@ export function unhideConditionalElements() {
     document.head.appendChild(styleEl);
     const conditionalEls = document.querySelectorAll('[data-visibility="conditional"]');
     for (const conditionalEl of conditionalEls) {
+        // In the case of a mega menu block, hide the mega menu link in the bar according to the same rule
+        // Change the visibility for Desktop and Mobile views
+        if (conditionalEl.classList.contains("s_mega_menu_odoo_menu")) {
+            const megaMenuDesktop = conditionalEl.closest("li[role~='presentation']");
+            const menusDesktop = megaMenuDesktop.parentElement.querySelectorAll('li');
+            // Get the index of the mega menu in the menu list
+            let index = -1;
+            menusDesktop.forEach((menuIter, i) => {
+                if (menuIter === megaMenuDesktop) {
+                    index = i;
+                }
+            });
+            const navDesktop = conditionalEl.closest("nav");
+            const navMobile = navDesktop.parentElement.querySelector('[aria-label="Mobile"]');
```

Comment:
Not robust enough selector:
1. The aria-label can be translated so this would work in english only
2. The could be another element with that aria-label

Reply:
According to the "menu.js" file I mentioned above, I used the class ".o_header_mobile" for the selector.